### PR TITLE
update licenses and notices on 7.3 dependencies

### PR DESCRIPTION
**Pull request by jsvd**
This PR is a consequence of running `./ci/license_check.sh` and fixing any unknown dependency licenses or missing NOTICEs.

### CODE REVIEW

**Comment by peterschretlen** [comment]
From the CI results it looks like the report is still not picking up the [aws v2 ruby sdk ](https://github.com/aws/aws-sdk-ruby/releases/tag/v2.11.331) license?

```
10:32:18 aws-sdk-core,2.11.331,,,UNKNOWN,
10:32:18 aws-sdk-resources,2.11.331,,,UNKNOWN,
10:32:18 aws-sdk,2.11.331,,,UNKNOWN,
```

**Comment by jsvd** [comment]
@peterschretlen true..the way we compile licenses is by building the logstash artifact and them checking if we have licenses for all the libraries..however we don't ping all libraries down so any updates to a library can cause this mismatch. libraries like aws are published once a day or more, so we keep getting out of sync.

The right fix is to stop verifying the notice file name down to the version number, but only to the library name.

So in summary the notice is there in logstash repo, the file name is aws-sdk....330-NOTICE.txt instead of 331

File: NOTICE.TXT [hunk 1]
```diff
@@ -1311,36 +1279,9 @@ copyright 2013. amazon web services, inc. all rights reserved.
    See the License for the specific language governing permissions and
    limitations under the License.
 
---------------------------------------------------
-Library: backports v3.11.3
```
Comment (danhermann) [inline NOTICE.TXT]: Looks like the notice for this library was not included in the generated file.

File: NOTICE.TXT [hunk 2]
```diff
@@ -1365,23 +1306,9 @@ LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
 OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
 SUCH DAMAGE.
 
---------------------------------------------------
-Library: bouncycastle v
```
Comment (danhermann) [inline NOTICE.TXT]: Also omitted?

**Comment by jsvd** [comment]
This is strange indeed, there are a bunch of libraries listed as "unused", which in turn got deleted from the NOTICE.txt

**Comment by jsvd** [comment]
licenses have been updated for all current branches: 6.8, 7.5, 7.6, 7.x and master
